src/stcal/ramp_fitting/likely_fit.py

```diff
@@ -0,0 +1,1353 @@
+from pprint import pprint
+
+import numpy as np
+
+from . import ramp_fit_class, utils
+from .likely_algo_classes import IntegInfo, Ramp_Result, Covar
+
+
+DELIM = "=" * 80
+SQRT2 = 1.41421356
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+BUFSIZE = 1024 * 300000  # 300Mb cache size for data section
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def likely_ramp_fit(
+    ramp_data, buffsize, save_opt, readnoise_2d, gain_2d, weighting, max_cores
+):
+    """
+    Setup the inputs to ols_ramp_fit with and without multiprocessing. The
+    inputs will be sliced into the number of cores that are being used for
+    multiprocessing. Because the data models cannot be pickled, only numpy
+    arrays are passed and returned as parameters to ols_ramp_fit.
+
+    Parameters
+    ----------
+    ramp_data : RampData
+        Input data necessary for computing ramp fitting.
+
+    buffsize : int
+        size of data section (buffer) in bytes (not used)
+
+    save_opt : bool
+       calculate optional fitting results
+
+    readnoise_2d : ndarray
+        readnoise for all pixels
+
+    gain_2d : ndarray
+        gain for all pixels
+
+    weighting : str
+        'optimal' specifies that optimal weighting should be used;
+         currently the only weighting supported.
+
+    max_cores : str
+        Number of cores to use for multiprocessing. If set to 'none' (the default),
+        then no multiprocessing will be done. The other allowable values are 'quarter',
+        'half', and 'all'. This is the fraction of cores to use for multi-proc. The
+        total number of cores includes the SMT cores (Hyper Threading for Intel).
+
+    Returns
+    -------
+    image_info : tuple
+        The tuple of computed ramp fitting arrays.
+
+    integ_info : tuple
+        The tuple of computed integration fitting arrays.
+
+    opt_info : tuple
+        The tuple of computed optional results arrays for fitting.
+    """
+    image_info, integ_info, opt_info = None, None, None
+
+    nints, ngroups, nrows, ncols = ramp_data.data.shape
+
+    if ngroups < 2:
+        raise ValueError("Likelihood fit requires at least 2 groups.")
+
+    readtimes = get_readtimes(ramp_data)
+
+    covar = Covar(readtimes, pedestal=False)  # XXX Choice of pedestal not given
+    integ_class = IntegInfo(nints, nrows, ncols)
+
+    readnoise_2d = readnoise_2d / SQRT2
+
+    for integ in range(nints):
+        data = ramp_data.data[integ, :, :, :]
+        gdq = ramp_data.groupdq[integ, :, :, :].copy()
+        pdq = ramp_data.pixeldq[:, :].copy()
+
+        # Eqn (5)
+        diff = (data[1:] - data[:-1]) / covar.delta_t[:, np.newaxis, np.newaxis]
+        alldiffs2use = np.ones(diff.shape, np.uint8)
+
+        for row in range(nrows):
+            d2use = determine_diffs2use(ramp_data, integ, row, diff)
+            d2use_copy = d2use.copy()  # Use to flag jumps
+            if ramp_data.nsig is not None:
```

Comment:
The variable name `nsig` was specifically requested.

Reply:
I know, but unless there is a good reason not to, I'd prefer to name the same concept with the same name everywhere it appears, as much as possible.  If we are going to (eventually) expose this parameter to the user, I'd like it to match the name in the existing jump step so that it's clear it's the same thing.